File: src/api/tauri/vault.ts
/**
 * Typed Tauri IPC wrappers for Recovery Vault commands.
 *
 * These are the ONLY entry points for Tauri `invoke` calls related to
 * vault entries. Components and pages MUST NOT call `invoke` directly
 * (doc 48 AC-FS-04).
 *
 * Command names are snake_case (Tauri convention).
 */

import { invoke } from '@tauri-apps/api/core';
import type { VaultEntry } from '../../types/device.types';

/** List vault entries for the local device, newest first. */
export const listVaultEntries = (): Promise<VaultEntry[]> =>
  invoke<VaultEntry[]>('list_vault_entries');

/** Create a system restore point (or platform equivalent record). */
export const createRestorePoint = (
  description?: string | null,
): Promise<VaultEntry> =>
  invoke<VaultEntry>('create_restore_point', {
    description: description ?? null,
  });

/** Export a DNA vault backup of device baseline data. */
export const createDnaVaultBackup = (): Promise<VaultEntry> =>
  invoke<VaultEntry>('create_dna_vault_backup');

/** Create a directory image from a source path. */
export const createDirectoryImage = (sourcePath: string): Promise<VaultEntry> =>
  invoke<VaultEntry>('create_directory_image', { sourcePath });

File: src-tauri/src/commands/vault.rs
//! Recovery Vault IPC.

use tauri::State;

use crate::error::CoreError;
use crate::models::VaultEntry;
use crate::vault;
use crate::AppState;

#[tauri::command]
pub fn list_vault_entries(state: State<'_, AppState>) -> Result<Vec<VaultEntry>, CoreError> {
    let conn = state.conn()?;
    vault::list_entries(&conn)
}

#[tauri::command]
pub fn create_restore_point(
    state: State<'_, AppState>,
    description: Option<String>,
) -> Result<VaultEntry, CoreError> {
    let conn = state.conn()?;
    vault::create_restore_point(&conn, description)
}

#[tauri::command]
pub fn create_dna_vault_backup(state: State<'_, AppState>) -> Result<VaultEntry, CoreError> {
    let mut conn = state.conn()?;
    vault::create_dna_vault_backup(&mut conn)
}

#[tauri::command]
pub fn create_directory_image(
    state: State<'_, AppState>,
    source_path: String,
) -> Result<VaultEntry, CoreError> {
    // File copy is heavy I/O; vault holds DB only for short insert at end.
    let conn = state.conn()?;
    vault::create_directory_image(&conn, source_path)
}
